frontend/src/components/TeamProject/TeamProjectSlice.ts

```diff
@@ -0,0 +1,176 @@
+  loading: boolean,
+  hasErrors: boolean,
+  project: TeamProjectState 
+}
+
+const initialState: initialstate = {
+  projectEditMode: false,
+  projectDeleteMode: false,
+  loading: false,
+  hasErrors: false,
+  project: {
+    _id: "loading...",
+    teamId: "loading...",
+    parentProjectId: "loading...",
+    projectName: "loading...",
+    projectUrl: "loading...",
+    description: "loading...",
+    mentor: "loading...",
+    referenceProjectName: "loading...",
+    sectionName: "loading..."
+  }
+};
+
+export const teamProjectSlice = createSlice({
+  name: 'teamProject',
+  initialState,
+  reducers: {
+    projectOperation: state => {
+      state.loading = true;
+    },
+    projectOperationSuccess: (state, { payload }) => {
+      state.project = payload;
+      state.loading = false;
+      state.hasErrors = false;
+    },
+    projectDeleteSuccess: (state) => {
+      state.loading = false;
+      state.hasErrors = false;
+    },
+    projectOperationFailure: state => {
+      state.loading = true;
+      state.hasErrors = true;
+    },
+    switchEditMode: state => {
+      state.projectEditMode = !state.projectEditMode;
+    },
+    switchDeleteMode: state => {
+      state.projectDeleteMode = !state.projectDeleteMode;
+    }
+  }
+});
+
+export const { 
+  projectOperation, 
+  projectOperationSuccess, 
+  projectDeleteSuccess,
+  projectOperationFailure, 
+  switchEditMode, 
+  switchDeleteMode 
+} = teamProjectSlice.actions;
+
+// if you want, add selectors here, change the one below, remember to register reducer in store.ts
+export const selectTeamProjects = (state: RootState) => state.teamProjects;
+
+export default teamProjectSlice.reducer;
+
+
+export function getProjectById(id: string, token: string) {
+  return async (dispatch: Dispatch) => {
+    dispatch(projectOperation())
+    try {
+      const responseProject = await request(
+      'GET', 
+      `https://coders-camp-organization-app.herokuapp.com/api/teams/projects/${id}`,
+      token
+      )
+      const data = await responseProject.json();
+      
+      const responseTeam = await request(
+        'GET', 
+        `https://coders-camp-organization-app.herokuapp.com/api/teams/${data.teamId}`,
+        token
+        )
+      const dataTeam = await responseTeam.json();
+      data.mentor = `${dataTeam.mentor.name} ${dataTeam.mentor.surname}`;
+
+      const responseParentProject = await request(
+        'GET', 
+        `https://coders-camp-organization-app.herokuapp.com/api/projects/${data.parentProjectId}`,
+        token
+        )
+      const dataParentProject = await responseParentProject.json();
+      data.referenceProjectName = dataParentProject.projectName;
+      
+      //... I don't know if this is meant to be done this way, seems kinda long :D
+      const responseSection = await request(
+        'GET', 
+        `https://coders-camp-organization-app.herokuapp.com/api/sections/${dataParentProject.sectionId}`,
+        token
+        )
```

Comment:
To make code cleaner use api.ts file in api folder:
```
import axios from 'axios';


export default axios.create({
  baseURL: process.env.REACT_APP_API_URL,
  timeout: 1000,
  headers: {
    'x-auth-token': localStorage.getItem('token')
  },
});
```
... set .env variable:
`REACT_APP_API_URL=https://coders-camp-organization-app.herokuapp.com/api`
..and then import this api in slice file
Sample usage:
`api.get('/projects/${data,parentProjectId}'`